docs/guides/latest/volumesnapshot/statefulset.md

```diff
@@ -0,0 +1,820 @@
+    apiGroup: null
+    kind: PersistentVolumeClaim
+    name: source-data-stash-demo-0
+status:
+  creationTime: "2019-07-15T07:59:14Z"
+  readyToUse: true
+  restoreSize: 1Gi
+```
+
+Here, `spec.snapshotContentName` field specifies the name of the `VolumeSnapshotContent` crd. It also represents the actual snapshot name that has been saved in Google Cloud. If we navigate to the `Snapshots` tab in the GCP console, we should see snapshot `snapcontent-6f3b49a9-a6d6-11e9-9f3a-42010a800050` has been stored successfully.
+
+<figure align="center">
+  <img alt="Stash Backup Flow" src="/docs/images/guides/latest/volumesnapshot/statefulset.png">
+<figcaption align="center">Fig: Snapshots in GCE Bucket</figcaption>
+</figure>
+
+## Restore PVC from VolumeSnapshot
+
+This section will show you how to restore PVCs from the snapshots we have taken in earlier section.
+
+**Create RestoreSession :**
+
+At first, we have to create a `RestoreSession` crd to restore PVCs from respective the snapshots.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: restore-pvc
+  namespace: demo
+spec:
+  driver: VolumeSnapshotter
+  target:
+    replicas : 3
+    volumeClaimTemplates:
+      - metadata:
+          name: restore-data-restore-demo-${POD_ORDINAL}
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: "standard"
+          resources:
+            requests:
+              storage: 1Gi
+          dataSource:
+            kind: VolumeSnapshot
+            name: source-data-stash-demo-${POD_ORDINAL}-1563177551
+#           name: source-data-stash-demo-0-1563177551
+            apiGroup: snapshot.storage.k8s.io
+```
+
+Here,
+
+- `spec.target.replicas`: `spec.target.replicas` specify the number of replicas of a StatefulSet whose volumes was backed up and Stash uses this field to dynamically create the desired number of PVCs and initialize them from respective or Specific VolumeSnapShots.
+- `spec.target.volumeClaimTemplates`:
+  - `metadata.name` is the name of the restored `PVC`.
+  - `spec.dataSource`: `spec.dataSource` specifies the source of the data from where the newly created PVC will be initialized. It requires following fields to be set:
+    - `apiGroup` is the group for resource being referenced. Now, Kubernetes supports only `snapshot.storage.k8s.io`.
+    - `kind` is resource of the kind being referenced. Now, Kubernetes supports only `VolumeSnapshot`.
+    - `name` is the `VolumeSnapshot` resource name. In `RestoreSession` crd, You must provide the name in the format:
+  
+    ```console
+     <VolumeSnapshot name prefix>-${POD_ORDINAL}-<backup session creation timestamp in Unix epoch seconds>
+     ```
+
+    `{POD_ORDINAL}` is resolved by the Stash operator. You can also restore all PVCs from a single VolumeSnapshot or restore each PVC from its respective VolumeSnapshot by setting up the name directly.
```

Comment:
The `${POD_ORDINAL}` variable is resolved by Stash. If you don't provide this variable and specify ordinal manually, all the PVC will be restored from the same VolumeSnapshot.